sklearn/cluster/optics_.py

```diff
@@ -386,6 +376,22 @@ def fit(self, X, y=None):
     # OPTICS helper functions
 
     def _calculate_optics_order(self, X, nbrs):
+        min_samples = self.min_samples
+
+        def _get_neighborhood(idx):
+            # TODO: special-case np.isinf(max_eps)
+            res = nbrs.radius_neighbors(X[idx:idx + 1], radius=self.max_eps)
+            (neigh_dist,), (neigh_idx,) = res
+            if len(neigh_idx) < self.min_samples:
+                core_dist = nbrs.kneighbors(X[point:point + 1],
```

Comment:
Not necessary. If there are less than `min_samples`, the value is *defined* to be "undefined", c.f. OPTICS paper.